### PR TITLE
MBox and Maildir added to account dialog

### PR DESCRIPTION
**Pull request by tikank**
I added widgets for file and directory selections for MBox and Maildir to account dialog. I also modified accountdialog.py to not refer directory to the imap/pop spesific members of the Account class. (Account class still contains those members since there are some other places which refers to them. I'll make another pull-request for that, when I have time for it...)

I still have some refactoring ideas for the account dialog code, that I would like to look later.

### CODE REVIEW

File: Mailnag/configuration/accountdialog.py [hunk 1]
```diff
@@ -39,6 +39,9 @@
 IDX_YAHOO	= 3
 IDX_IMAP	= 4
 IDX_POP3	= 5
+IDX_MBOX	= 6
+IDX_MAILDIR	= 7
+
 
 PROVIDER_CONFIGS = [
 	[ 'Gmail', 'imap.gmail.com', '993'],
```
Comment (tikank) [inline Mailnag/configuration/accountdialog.py]: Refactoring idea: These provider configs could be replaced by adding them to the backends map with suitable defaults in `Mailnag/backends/__init__.py`. They all could use IMAPMailboxBackend class. Then account dialog does not need to handle them specially.

File: Mailnag/configuration/accountdialog.py [hunk 2]
```diff
@@ -68,12 +71,18 @@ def __init__(self, parent, acc):
...
+		self._label_account_password = builder.get_object("label_account_password")
 		self._entry_account_password = builder.get_object("entry_account_password")
 		self._label_account_server = builder.get_object("label_account_server")
 		self._entry_account_server = builder.get_object("entry_account_server")		
 		self._label_account_port = builder.get_object("label_account_port")
 		self._entry_account_port = builder.get_object("entry_account_port")
+		self._label_account_file_path = builder.get_object("label_account_file_path")
+		self._chooser_account_file_path = builder.get_object("chooser_file_path")
+		self._label_account_directory_path = builder.get_object("label_account_directory_path")
+		self._chooser_account_directory_path = builder.get_object("chooser_directory_path")
```
Comment (tikank) [inline Mailnag/configuration/accountdialog.py]: Refactoring idea: All parameter have two widgets, a label and some kind of entry widget (entry, checkbox, chooser), and all these pairs are handled similar ways. The pairs could be wrapped to some kind of parameter objects which all have identical inteface, something like:
```
self._user_entry = EntryParam(builder, "label_account_user", "entry_account_user")
self._file_path_entry = ChooserParam(builder, "label_account_file_path", "chooser_file_path")
...
```
Then all the *Param classes have methods set_value, get_value, set_visible, etc.

I can't explain this better yet, I had to think this a bit more...

**Comment by pulb** [comment]
Again many thanks for all your work. I'm currently busy mastering some real life problems so it will take some time to get this reviewed and merged. This shouldn't stop you from submitting further pull requests though :grin:

File: Mailnag/common/accounts.py [hunk 3]
```diff
@@ -47,25 +47,32 @@
...
+		self.password = config.get('password', '')
+		self.oauth2string = config.get('oauth2string', '')
+		self.server = config.get('server', '')
+		self.port = config.get('port', '')
+		self.ssl = config.get('ssl', True)
+		self.imap = config.get('imap', True)
+		self.idle = config.get('idle', False)
+		self.folders = config.get('folders', [])
+		self._rest_of_config = config
 		self._backend = None
```
Comment (pulb) [inline Mailnag/common/accounts.py]: I guess you should call self._backend.close() (if it is not None) before setting it to None

Comment (tikank) [inline Mailnag/common/accounts.py]: Yes, that would be wise (as long as close exists, see https://github.com/pulb/mailnag/pull/132#issuecomment-248082128).

**Comment by pulb** [comment]
Sorry for the delay. I moved to another city and currently only have temp internet access. I will continue working on mailnag as soon as things get back to normal :)

**Comment by pulb** [comment]
Thanks, finally merged :-)

**Pull request merged**